src/main/java/edu/ucsb/cs156/rec/services/RequestTypeService.java

```diff
@@ -0,0 +1,61 @@
+package edu.ucsb.cs156.rec.services;
+
+import edu.ucsb.cs156.rec.entities.RequestType;
+import edu.ucsb.cs156.rec.errors.EntityAlreadyExistsException;
+import edu.ucsb.cs156.rec.repositories.RequestTypeRepository;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+/**
+ * This is a service that provides information about the current user.
+ * 
+ * This is the version of the service used in production.
+ */
+
+@Slf4j
+@Service("RequestType")
+public class RequestTypeService {
+
+	@Autowired
+	private RequestTypeRepository requestTypeRepository;
+
+	private boolean requestTypeExists(String type, Iterable<RequestType> typeList) {
+		for (RequestType requestType : typeList) {
+			if (requestType.getRequestType().equals(type)) {
+				return true;
+			}
+		}
+		return false;
+	}
+
+	public RequestType trySave(RequestType requestType) throws EntityAlreadyExistsException {
```

Comment:
You don't need to get ALL of the request types and iterate through them.  Let the database do the work for you.

Instead, add a method to the RequestTypeRepository called 

```java
  Optional<RequestType> findByRequestType(String requestType);
```

Just look up the RequestType and see if it exists.  If it does, it's a duplicate. If not, it isn't.

Any time you are "looping through database records" looking for something, you should stop and say: wait... there's probably a better way.